lib/services.rb

```diff
@@ -57,7 +62,44 @@ def env_local_file
     ENV["LOG_DIR"] || default
   end
 
+  Services.register(:redirect_file) do
+    yyyymm = DateTime.now.strftime "%Y%m"
+    default_file = "/htapps/babel/hathifiles/catalog_redirects/redirects/redirects_#{yyyymm}.txt.gz"
+    # Start migrating from redirect_file to REDIRECT_FILE on principle of least surprise
+    ENV["redirect_file"] || ENV["REDIRECT_FILE"] || default_file
+  end
+
   Services.register(:db) do
-    DBH.connect
+    Sequel.connect(Services[:db_connection_string], login_timeout: 2, pool_timeout: 10, max_connections: 6)
+  end
+
+  Services.register(:db_connection_string) do
+    "jdbc:mysql://#{ENV["MYSQL_HOST"]}/#{ENV["MYSQL_DATABASE"]}? \
+    user=#{ENV["MYSQL_USER"]}&password=#{ENV["MYSQL_PASSWORD"]}& \
+    useTimezone=true&serverTimezone=UTC"
+  end
+
+  Services.register(:no_db?) do
+    ENV["NO_DB"] == "1" or Services[:no_external_data?]
```

Comment:
I would be inclined to check this at startup rather than deferring via `Services`, and in the tests explicitly re-register the mock object if we need one; I would also be inclined to check for presence/truthiness rather than require the specific value of "1".

```ruby
if ENV["NO_DB"] || ENV["HT_NO_EXTERNAL_DATA"]
  Services.register(:print_holdings) { MockPrintHoldings.new }
  Services.register(:oclc_resolution) { MockOCLCResolution.new }
else 
  Services.register(:print_holdings) { PrintHoldings.new }
  Services.register(:oclc_resolution) { OCLCResolution.new }
end

if ENV["NO_REDIRECTS"] || ENV["HT_NO_EXTERNAL_DATA"]
  Services.register(:mock_redirects) { MockRedirects.new }
else
  Services.register(:redirects) { Redirects.new }
end
```

Reply:
Actually I changed my mind on this, ignore the above comments...